fenicsadapter/fenicsadapter.py

```diff
@@ -1,759 +1,458 @@
-                vertices_x.append(v.x(0))
-                if self._dimensions == 2:
-                    vertices_y.append(v.x(1))
-                elif self._can_apply_2d_3d_coupling():
-                    vertices_y.append(v.x(1))
-                    vertices_z.append(0)
-                else:
-                    raise Exception("Dimensions do not match!")
-
-        assert(n != 0), "No coupling boundary vertices detected"
-
-        if self._dimensions == 2:
-            return fenics_vertices, np.stack([vertices_x, vertices_y], axis=1), n
-        elif self._dimensions == 3:
-            return fenics_vertices, np.stack([vertices_x, vertices_y, vertices_z], axis=1), n
-
-    def _are_connected_by_edge(self, v1, v2):
-        """Returns true if both vertices are connected by an edge. """
-        for edge1 in dolfin.edges(v1):
-            for edge2 in dolfin.edges(v2):
-                if edge1.index() == edge2.index():  # Vertices are connected by edge
-                    return True
-        return False
-
-    def _extract_coupling_boundary_edges(self, id_mapping):
-        """Extracts edges of mesh which lie on the boundary.
-        :return: two arrays of vertex IDs. Array 1 consists of first points of all edges
-        and Array 2 consists of second points of all edges
-
-        NOTE: Edge calculation is only relevant in 2D cases.
-        """
-
-        vertices = dict()
-
-        for v1 in dolfin.vertices(self._mesh_fenics):
-            if self._coupling_subdomain.inside(v1.point(), True):
-                vertices[v1] = []
-
-        for v1 in vertices.keys():
-            for v2 in vertices.keys():
-                if self._are_connected_by_edge(v1, v2):
-                    vertices[v1] = v2
-                    vertices[v2] = v1
-
-        vertices1_ids = []
-        vertices2_ids = []
-
-        for v1, v2 in vertices.items():
-            if v1 is not v2:
-                vertices1_ids.append(id_mapping[v1.global_index()])
-                vertices2_ids.append(id_mapping[v2.global_index()])
-
-        vertices1_ids = np.array(vertices1_ids)
-        vertices2_ids = np.array(vertices2_ids)
-
-        return vertices1_ids, vertices2_ids
-
-    def set_coupling_mesh(self, mesh, subdomain, use_nearest_projection=True):  # as soon as issue https://github.com/precice/fenics-adapter/issues/53 is fixed change default to use_nearest_projection=True
-        """Sets the coupling mesh. Called by initalize() function at the
-        beginning of the simulation.
-        """
-        self._coupling_subdomain = subdomain
-        self._mesh_fenics = mesh
-        self._fenics_vertices, self._coupling_mesh_vertices, self._n_vertices = self._extract_coupling_boundary_vertices()
-        self._vertex_ids = self._interface.set_mesh_vertices(self._mesh_id, self._coupling_mesh_vertices)
-
-        """ Define a mapping between coupling vertices and their IDs in precice"""
+        """ Set up mesh in preCICE """
```

Comment:
```suggestion
        # Set up mesh in preCICE
```
Why not comment like this? `"""` is reserved for [docstrings](https://www.python.org/dev/peps/pep-0008/#documentation-strings). There is a nice explanation of docstrings vs. comments [here](https://stackoverflow.com/a/19076001/5158031). I would say this comment above is actually a comment and not a docstring. Note that for a user of the FEniCS adapter the preCICE API calls are not relevant, since they are supposed to be hidden inside the adapter.